.github/workflows/build-crate-and-npm.yml

```diff
@@ -0,0 +1,81 @@
+name: Build crate and npm package
+
+on:
+  push:
+    branches:
+      - main
+      - dev
+    tags:
+        - '**'
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  build_crate:
+    name: Crate
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Set up rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        components: rustfmt, clippy
+
+    - name: Set up rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: nightly
+        components: rustfmt, clippy
+
+    - name: fmt
+      run: cargo +nightly fmt -- --files-with-diff --check
+    - name: clippy
+      run: cargo +nightly clippy
```

Comment:
we already have a workflow that runs `fmt` and `clippy`, right? is there a good reason to add another one?

Reply:
Mostly it's so that if there's a problem, it won't upload a half-baked version to `crates.io` or `npmjs.com` (which isn't yet implemented anyway).